apps/api/src/campaign/campaign.controller.ts

```diff
@@ -167,6 +173,83 @@ export class CampaignController {
       )
   }
 
+  @Post(':id/subscribe')
+  @Public()
+  async subscribeToCampaignNotifications(
+    @Param('id') id: string,
+    @Body() data: CampaignSubscribeDto,
+  ) {
```

Comment:
this function looks like quite long for to be in a controller - is it possible to break down and move the specific parts to the specific service for better encapsulation?